src/lib/ipr.js

```diff
@@ -136,29 +137,34 @@ function createIPR (params) {
   *
   * @param {Object} plugin Ledger plugin to listen on
   * @param {Object} params Parameters for creating payment request
-  * @param {Buffer} params.secret Secret to generate fulfillments with
+  * @param {Buffer} params.receiverSecret Secret to generate fulfillments with
   * @param {Buffer} [params.allowOverPayment=true] Accept payments with higher amounts than expected
   * @param {IncomingCallback} callback Called after an incoming payment is validated.
   *
   * @return {Object} Payment request
   */
-function listen (plugin, params, callback) {
-  return Transport.listen(plugin, params, callback, 'ipr')
+async function listen (plugin, params, callback) {
+  const address = (await ILDCP.fetch(plugin.sendData.bind(plugin))).clientAddress
```

Comment:
`plugin.sendData.bind` will fail for v1 plugins since `plugin` hasn't been wrapped by `ilp-compat-plugin`.